src/libos/src/fs/rootfs.rs

```diff
@@ -140,7 +142,24 @@ pub fn mount_nonroot_fs_according_to(
                 mount_fs_at(ramfs, root, &mc.target, follow_symlink)?;
             }
             TYPE_DEVFS => {
-                let devfs = dev_fs::init_devfs()?;
+                let disk_options = {
+                    // Setup SwornDisk before adding to DevFS if enabled
+                    let ext2_mc_opt = mount_configs
+                        .iter()
+                        .find(|mc| mc.type_ == ConfigMountFsType::TYPE_EXT2);
+                    if let Some(mc) = ext2_mc_opt {
+                        let disk_size = mc.options.disk_size;
+                        if disk_size.is_none() {
+                            return_errno!(EINVAL, "Disk size is expected for Ext2");
+                        }
+                        let source_path = mc.source.as_ref();
+                        SwornDiskMeta::setup(disk_size.unwrap(), user_key, source_path)?;
```

Comment:
I think `SwornDiskMeta::setup `should be moved to `DiskAsINode::open_or_create`.
Because it is not guaranteed that `Devfs` will be mounted before `Ext2`.

Reply:
Now I move the "disk metadata setup" to the first step in `mount_nonroot_fs_according_to(...)`, and move the "Ext2 mount" to the final step.